iron-form.html

```diff
@@ -135,27 +158,25 @@
       */
 
       attached: function() {
-        this._nodeObserver = Polymer.dom(this.$.content).observeNodes(
```

Comment:
This PR is kind of enormous and hard to review, as a result. I would have preferred one PR that just fixes `reset`,  and one that fixes the slots (especially since I'm not convinced we want to complicate this already horrendous element with that) 😭